### DISCUSSION TITLE
Custom Note Template

### DISCUSSION THREAD

**Discussion by renarl** [davish/obsidian-full-calendar#396] [Feature Requests]
I want to use a custom Templater template for new notes.

Specifically to add extra metadata fields, like a project, participants, etc., and also to have some frequently used headings and tags.

**Comment by davish** [comment]
This is planned and mentioned in the README, but somehow I never created an issue for it. I'll use this issue to track this feature, thanks for submitting @renarl!

**Comment by s0tho** [comment]
I was checking to see whether my idea had already been proposed or not and this issue seems pretty close. 

I'll just add that it'd be nice if we could add projects tag like `#project/this-is-a-project` and give it a color (or base the color on any metadata). I know this can technically be done by creating one calendar for each project but it seems to be a bit overkilling and a bother since you have to manually setup a new calendar for each project. 

Now, having multiple project in one calendar could lead to collisions. Though if you want to avoid collision between files, you could always add an option to choose the saving format (timestamp, date + Name, ...)

Thanks for this plugin btw, I was looking for a FOSS/local alternative to [toggl](https://track.toggl.com) and didn't think to look into obsidian's plugins before today

**Comment by mayurankv** [comment]
I too would love the option to have a custom template that would ideally work with Templater optionally. In addition, it would be amazing if you could set alternate metadata fields for this plugin. I use the `type` field across all my notes and thus this would really interfere with using events created by this plugin as notes as well. It would be great to be able to refactor the metadata under a single nested value i.e.:
```
Full_Calendar:
  Start_Date: 
  End_Date: 
  Recurring: 
  ...
```
I also love the idea of being able to set a colour based on some metadata (ideally for me that would based on evaluation of a dvjs query).

**Reply by jreus** [reply]
This is a great idea for Full Calendar specific properties.
But I'm actually rather happy to have things like startDate, endDate, startTime, endTime etc... be general properties that can also be consumed by other plugins.

**Reply by mayurankv** [reply]
Sure, I think it would be great if FullCalendar used that value by default but if a different setting is set via Full_Calendar: StartDate:, then it should use that value.

**Comment by MGOPW** [comment]
+1 - Also looking for this feature right now, i would like for it to use the template for my meeting notes that I already have set up in my obsidian vault.

**Comment by igelbox** [comment]
After quick research it appears to be a Templater issue coz it [ignores](https://github.com/SilentVoid13/Templater/blob/c461524bde5deda2dabeb15e93ed0e697d94b7a9/src/core/Templater.ts#L425) folder template when file isn't empty and uses the file content as a template instead. So I see a couple of ways from here:
- ask Templater devs to implement a checkbox to forcibly apply folder templates even if there's some content in newly created file (however I'm not sure how much the current behavior is desirable by Templater itself)
- implement a way to set a note template in Full Calendar plugin settings - I can prove with #283 that Templater post-processes this note content automatically just after note creation (which seems like desired behavior here)

**Comment by elvarb** [comment]
I'm not sure this is directly related but for meeting notes I am using quickadd + templater to create a meeting note template which also creates a link to the meeting in my daily note.

I was working on modifying the template and if I add anything else than what Full Calendar wants in the frontmatter it will not display in the calendar view.

For example what works
```
---
title: Meeting title
allDay: false
startTime: 11:00
endTime: 12:00
date: 2023-03-13
completed: null
---
```

And what does not work
```
---
title: Meeting title
allDay: false
startTime: 11:00
endTime: 12:00
date: 2023-03-13
completed: null
type: meeting
---
```

**Comment by davish** [comment]
Hi @elvarb, `type` is a reserved frontmatter keyword for full calendar events. If it is not present, it defaults to `single`. Right now the accepted values are `single` and `recurring`. I'd suggest calling your custom field `eventType`, or something to that effect. That should solve the issue.

**Reply by elvarb** [reply]
As to avoid this conflict and future conflict with other plugins, would it be possible to customize the reserved frontmatter keywords for full calendar?

Field name such as type is too generic to be reserved for a single plugin.

**Reply by jreus** [reply]
+1 for this. I've been working on templating solutions for quickly creating Full Calendar notes that also play well with other systems. `type` has already conflicted for me once so far with another plugin.. since this is basically a flag for whether or not the event is recurring, why not name the property `recurring` and give it a bool value?

Or, if you'd like to keep the idea of a flexible multi-value type.. `eventType`?

**Comment by mayurankv** [comment]
Is there any likelihood of making the scheme customisable? i.e. `type` could be changed by the user to `fullcalendar_type`. Or as I suggested above some nesting of the metadata within a `full_calendar` key to get a nested YAML dictionary? People could then continue using the `type` key for their own purposes (which I'm sure many people, alongside myself, use).

**Comment by Tanjamuse** [comment]
Either I haven't found the setting or it haven't been made possible yet, because I can't figure out how to make Full Calendar make a note depending on a templater template.
